### PR TITLE
Move more env variables to shared memory

### PR DESCRIPTION
**Pull request by sporksmith**
Progress on https://github.com/shadow/shadow/issues/2848 and on refactoring process initialization and teardown.

### CODE REVIEW

File: src/lib/shadow-shim-helper-rs/src/shim_shmem.rs [hunk 1]
```diff
@@ -79,6 +79,9 @@ pub struct HostShmem {
     // per-process option.
     pub unblocked_vdso_latency: SimulationTime,
 
+    // Native pid of the Shadow process.
+    pub shadow_pid: libc::pid_t,
```
Comment (stevenengler) [inline src/lib/shadow-shim-helper-rs/src/shim_shmem.rs:83]: How about "the Shadow simulator process"? When I first read "Shadow process", I thought it meant "Shadow `Process`" and was confused.

File: src/lib/shim/shim.c [hunk 2]
```diff
@@ -228,41 +228,21 @@ static void _shim_parent_init_logging() {
...
-    // Ensure that the child process exits when Shadow does.  Shadow
-    // ought to have already tried to terminate the child via SIGTERM
-    // before shutting down (though see
-    // https://github.com/shadow/shadow/issues/903), so now we jump all
-    // the way to SIGKILL.
+    // Ensure that the child process exits when Shadow does. This is to avoid
+    // confusing behavior or a "stalled out" process in the case that Shadow
+    // exits abnormally. Shadow normally ensures all managed processes have
+    // exited before exiting itself.
     if (prctl(PR_SET_PDEATHSIG, SIGKILL) < 0) {
```
Comment (stevenengler) [inline src/lib/shim/shim.c:236]: Is there a way we can set this with posix_spawn, for example if someone tries to run a static binary where this function isn't loaded/run?

**Review by stevenengler — APPROVED** [review]

File: src/lib/shadow-shim-helper-rs/src/shim_shmem.rs (hunk 1 above)
Comment (sporksmith) [reply src/lib/shadow-shim-helper-rs/src/shim_shmem.rs]: Done

File: src/lib/shim/shim.c (hunk 2 above)
Comment (sporksmith) [reply src/lib/shim/shim.c]: Unfortunately not. Added a comment.

**Pull request merged**